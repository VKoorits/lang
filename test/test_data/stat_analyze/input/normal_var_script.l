sub factorial(n):
	return 1
		
		
var declared_var = 42
var d, index = declared_var, name = "Victor" 
var string = "hello" + "world", num = factorial(5)
var lst = [1,2,3,{"k" => string}, 8, string, num, factorial(4)]

declared_var //=2
index = 0
d = "D"
string = "hello, Victor"
num = 24
lst = []
name = "victOr"
