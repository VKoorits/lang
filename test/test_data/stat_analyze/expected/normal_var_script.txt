INIT_0 INIT_1 INIT_2 INIT_3 declared_var 2 //= index 0 = d D = string hello, Victor = num 24 = lst ARR_4 = name victOr = NULL
INIT_0 :declared_var ELEM_5 NULL
ELEM_5 :42 NULL
INIT_1 :d index ELEM_5 name ELEM_6 NULL
ELEM_5 :declared_var NULL
ELEM_6 :Victor NULL
INIT_2 :string ELEM_5 num ELEM_6 NULL
ELEM_5 :hello world + NULL
ELEM_6 :CALL_7 NULL
CALL_7 :LIST_8 factorial NULL
LIST_8 :ELEM_9 NULL
ELEM_9 :5 NULL
INIT_3 :lst ELEM_5 NULL
ELEM_5 :ARR_6 NULL
ARR_6 :ELEM_7 ELEM_8 ELEM_9 ELEM_10 ELEM_11 ELEM_12 ELEM_13 CALL_14 NULL
ELEM_7 :1 NULL
ELEM_8 :2 NULL
ELEM_9 :3 NULL
ELEM_10 :HASH_15 NULL
HASH_15 :ELEM_16 ELEM_17 NULL
ELEM_16 :k NULL
ELEM_17 :string NULL
ELEM_11 :8 NULL
ELEM_12 :string NULL
ELEM_13 :num NULL
CALL_14 :LIST_15 factorial NULL
LIST_15 :ELEM_16 NULL
ELEM_16 :4 NULL
ARR_4 :ELEM_5 NULL
ELEM_5 :NULL
